### ISSUE TITLE
App won't connect to mqtt broker on HomeAssistant

### ISSUE THREAD

**Issue by Mariusthvdb** [owntracks/ios#568]
HI,

all of a sudden my App won't connect anymore to my unchanged setup of mqtt broker on a Homeassistant instance.

in settings MQTT mode, trying to connect securely on port 8883 proto 4 tls enabled with the correct user id and password (all my other mqtt connections run smoothly..) i receive his error:

status

```
idle The operation couldn't be completed.
(kCFErrorDomainCFNetwork error 1.) {
"_kCFStreamErrorCodeKey" = 1,
"_kCFStreamErrorDomainKey" = 12;
}
```

not sure what to do, please help?

**Comment by jpmens** [comment]
I bet your MQTT broker *has* changed on your instance: its TLS certificate has expired. Possible?

**Comment by Mariusthvdb** [comment]
wouldn't now how to check ;-)

but my duckdns certs are fine:

![schermafbeelding 2019-02-07 om 12 16 16](https://user-images.githubusercontent.com/33354141/52408062-3f98fb80-2ad2-11e9-83a9-a7e4e9a2ebe6.jpg)

**Comment by jpmens** [comment]
Does the hostname you are using still resolve to an IP address?

**Comment by Mariusthvdb** [comment]
Yes No issues there. I see the instance in my HA app and all MQTT connections are just fine.

Was wondering if we had an update of the OwnTracks app lately that might cause this?

**Comment by jpmens** [comment]
kCFErrorDomainCFNetwork = 1 is host not found as per the Apple dev documentation.

**Comment by Mariusthvdb** [comment]
i don't understand why it would say that...
to be sure: I have entered this:

Host: https://mydomain.duckdns.org
Port: 8883
UserID: mqtt userid
Password: mqtt password
Secret encryption key: #empty

tried it without the https:// but that errors out too

**Comment by jpmens** [comment]
You said "settings MQTT mode", so what is `https://` doing in youir Host field, in particular with port `8883`? 

Are you doing MQTT or are you doing HTTP?

**Comment by Mariusthvdb** [comment]
yes, on the settings page in the iOS Owntracks app, I need to enter these?

my mqtt broker resides on https://mydomain.duckdns.org:8883

have had that ever since, and was working alright, should it be configured differently then?

**Comment by jpmens** [comment]
Your MQTT broker does definitely not reside at that address, and that has never worked. Please read the documentation.

**Comment by jpmens** [comment]
Hint: MQTT (the protocol) is not the same as HTTPS (the protocol)

**Comment by Mariusthvdb** [comment]
no need to start being sarcastic here... ;-)
really sorry, should have explained that my domain.duckdns.org:8123 is my mqtt hub, on which the mqtt broker is configured to use secure communication via port 8883

it did work, and my sensors all have this format:

```
  - platform: mqtt
    state_topic: !secret owntracks_mqtt_topic
    name: 'Ot Longitude'
    icon: mdi:map-marker-radius
    value_template: '{{value_json.lon | round(3)}}'
```

proof of the connection and correct settings.

I might have mixed things up now, because HomeAssistant has changed the way it configures Owntracks integration over http, but I would like to keep using mqtt.

**Comment by jpmens** [comment]
I am not being sarcastic. I'm telling you that never worked.

If you are connecting via the MQTT protocol, then your host is `domain.duckdns.org` and your port is `8883`and `TLS` is True.

FYI, neither on port 8123 nor on port 8883 is there something TCP'ish listening, at least I cannot open a TCP connection to those.

**Comment by Mariusthvdb** [comment]
> If you are connecting via the MQTT protocol, then your host is `domain.duckdns.org` and your port is `8883`and `TLS` is True.

thanks. im am trying to do that. But, on domain.duckdns.org 2 HA instances reside. I connect to the Mqtt instance on https://mydomain.duckdns.org:8123, and the other instance listens on another port. I forward these ports in my router to the correct ip addresses.

Now how can I set that in the Owntracks app?

It really used to work, I have got no idea why this has stopped all of a sudden...
